src/main/java/org/opentripplanner/routing/graph/GraphIndex.java

```diff
@@ -54,6 +56,8 @@
     private final Map<Station, MultiModalStation> multiModalStationForStations = Maps.newHashMap();
     private final HashGridSpatialIndex<TransitStopVertex> stopSpatialIndex = new HashGridSpatialIndex<>();
     private final Map<ServiceDate, TIntSet> serviceCodesRunningForDate = new HashMap<>();
+    private final Map<FeedScopedId, TripOnServiceDate> tripOnServiceDateById = new HashMap<>();
+    private final Map<T2<FeedScopedId, ServiceDate>, TripOnServiceDate> tripOnServiceDateForTripAndDay = new HashMap<>();
```

Comment:
We should decide if we want to use `TripIdAndServiceDate`  or `T2<FeedScopedId, ServiceDate>`, and use it consistently everywhere

Reply:
I vote for using named tuples, that is creating classes where needed - at least where they are part of the api for a module/package. Internally in a class (and maybe between classes inside a package) it is ok to use the T2. I am not sure if we want to delay this PR any more. So, maybe we can create an issue for it and take it as part of cleanup.